### PR TITLE
SNOW-1658914: init record_meta column type

### PR DESCRIPTION
**Pull request by sfc-gh-wtrefon**
# Overview

SNOW-1658914

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java [hunk 1]
```diff
@@ -158,6 +162,37 @@ public void createTableWithOnlyMetadataColumn(final String tableName) {
     LOGGER.info("Created table {} with only RECORD_METADATA column", tableName);
   }
 
+  @Override
+  public void initializeMetadataColumnTypeForIceberg(String tableName) {
+    checkConnection();
+    InternalUtils.assertNotEmpty("tableName", tableName);
+    String query =
+        "ALTER ICEBERG TABLE identifier(?) ALTER COLUMN RECORD_METADATA SET DATA TYPE OBJECT("
+            + "offset INTEGER,"
```
Comment (sfc-gh-wtrefon) [inline src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:171]: double-check if the list of properties is correct and the data types

Comment (sfc-gh-mbobowski) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:171]: Looks good, but I wonder if we should add `LogAppendTime BIGINT` as well. I found the following code
```
// ignore if no timestamp
    if (record.timestampType() != TimestampType.NO_TIMESTAMP_TYPE
        && metadataConfig.createtimeFlag) {
      meta.put(record.timestampType().name, record.timestamp());
    }
```

and there are also tests like `public void testTimeStamp() throws IOException {`

File: src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  @Test
+  void shouldThrowExceptionWhenTableDoesNotExist() {
+    assertThatThrownBy(() -> icebergInitService.initializeIcebergTableProperties(tableName))
+        .isInstanceOf(SnowflakeKafkaConnectorException.class);
+  }
+
+  @Test
+  void shouldThrowExceptionWhenRecordMetadataDoesNotExist() {
```
Comment (sfc-gh-bzabek) [inline src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java:66]: I feel dumb. What is tested here?
We `ALTER ICEBERG TABLE identifier(?) ALTER COLUMN RECORD_METADATA ` in IcebergInitService... and also throw an error when there is no record_metadata ??

Comment (sfc-gh-bzabek) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java:66]: By the way you can also verify exception code. Will be more readable.
```
    assertThatThrownBy(() -> JdbcProperties.create(connection, proxy, jdbcMap))
        .isInstanceOfSatisfying(
            SnowflakeKafkaConnectorException.class,
            ex -> {
              // property key is printed not value
              assertTrue(ex.getMessage().contains("user"));
              assertEquals("0031", ex.getCode());
            });
```

Comment (sfc-gh-bzabek) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java:66]:
```
net.snowflake.client.jdbc.SnowflakeSQLException: SQL compilation error: error line 1 at position 47
invalid identifier 'RECORD_METADATA'
```
I don't get it.

File: src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    tableName = TestUtils.randomTableName();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    dropIcebergTable(tableName);
+  }
+
+  @Test
+  void shouldInitializeMetadataType() {
```
Comment (sfc-gh-bzabek) [inline src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java:37]: And here we create iceberg table with "record_metadata" column and then run a query to alter it with "record_metadata". How doeas it work?

File: src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java (hunk 2 above)
Comment (sfc-gh-bzabek) [reply src/test/java/com/snowflake/kafka/connector/streaming/iceberg/IcebergInitServiceIT.java:66]: aa I see we throw an error that there is no record_metadata column, when sql query throws an error...

File: src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java (hunk 1 above)
Comment (sfc-gh-wtrefon) [reply src/main/java/com/snowflake/kafka/connector/internal/SnowflakeConnectionServiceV1.java:171]: done

**Review by sfc-gh-mbobowski — APPROVED** [review]

**Pull request merged**